test/test_framework.py

```diff
@@ -30,13 +30,16 @@
 
 import ops
 from ops.framework import _BREAKPOINT_WELCOME_MESSAGE, _event_regex
+from ops.jujucontext import _JujuContext
 from ops.model import _ModelBackend
 from ops.storage import JujuStorage, NoSnapshotError, SQLiteStorage
 from test.test_helpers import FakeScript
 
 
 def create_model():
     """Create a Model object."""
+    if 'JUJU_VERSION' not in os.environ:
+        os.environ['JUJU_VERSION'] = '0.0.0'
```

Comment:
@IronCore864 why does the test framework need to manipulate the environment now?  The new context appears to already pick a default if JUJU_VERSION is not present.

cc @gabrielcocenza

Reply:
> @IronCore864 why does the test framework need to manipulate the environment now? The new context appears to already pick a default if JUJU_VERSION is not present.

Note that this is the internal tests for ops themselves, not ops.testing (Harness). It seems like no-one else should really care what the internal ops tests are doing. Does this actually impact you, or is it just something you noticed while running the ops tests or similar?

It's being set here because `_JujuContext` expects the environment variable to be set, because all supported Juju versions so do.

Reply:
> Does this actually impact you, or is it just something you noticed while running the ops tests or similar?

Yes, it's affecting out unit tests that are checking the arguments that were called in subprocess. See an example at canonical/charm-simple-streams#22

Reply:
Ah, it's not the code linked above (your tests will never run that), it's [this similar code in ops.testing](https://github.com/canonical/operator/blob/cdfd10f34f2d8338fb18dea1185cdd70ea082252/ops/testing.py#L263).

We failed to consider this :disappointed:, and we must not have run this change against the full set of charms we know about :disappointed:. It would indeed be better to have Harness not change the environment. It's not immediately clear to me what we should do now: we could change the behaviour back so that it doesn't, but if everyone that this impacts has then already changed their code by the time that fix gets out, then it's just doubling up the impact (although it is cleaner in general).

I'll talk with the team about this later today.

Reply:
FYI, we're going to do an ops 2.16.1 change that reverts this behaviour.

Reply:
I confirm that now it's fixed :+1: